internals/updates/docker.go

```diff
@@ -52,17 +62,120 @@ func (du *DockerUpdaterCleaner) Clean(ctx context.Context) error {
 					}
 				}
 			}
+
+			log.Debug().Strs("container_name", c.Names).
+				Str("state", c.State).
+				Str("id", c.ID).
+				Msg("Agent removes the updater container successfully")
+
 		} else if c.State == "running" {
 			foundRunningContainer = true
 		}
 	}
 
 	if foundRunningContainer {
-		return errors.New("Found running updater container. Retry after 30 seconds.")
+		return errors.New("found running updater container. Retry after 30 seconds")
 	}
 	return nil
 }
 
 func (du *DockerUpdaterCleaner) UpdateID() int {
 	return du.updateID
 }
+
+func updateAgentInfoIfNeeds(ctx context.Context, updateID int) error {
+	cli, err := docker.NewClient()
+	if err != nil {
+		return err
+	}
+	defer cli.Close()
+
+	possibleImagePrefixes := []string{"portainer/agent", "portainerci/agent"}
```

Comment:
You cannot use a predefined list here. Portainer server has a `EDGE_UPDATE_AGENT_IMAGE_PREFIX ` and `EDGE_UPDATE_AGENT_IMAGE_VERSION` env vars that can be used to deploy any image name